src/crush/CrushWrapper.cc

```diff
@@ -3131,7 +3131,21 @@ class CrushTreeFormattingDumper : public CrushTreeDumper::FormattingDumper {
     f->open_array_section("nodes");
     Parent::dump(f);
     f->close_section();
+
+    // There is no stray bucket whose id is a negative number, so just get
+    // the max_id and iterate from 0 to max_id to dump stray osds.
     f->open_array_section("stray");
+    auto &name_map = this->crush->name_map;
+    int32_t max_id = -1;
+    if (!name_map.empty()) {
+        max_id = name_map.rbegin()->first;
+    }
+    for (int32_t i = 0; i <= max_id; i++) {
+        if (this->crush->item_exists(i) &&
```

Comment:
wrong indent. and please drop `this->` , i don't think we need to follow how `python` passes `this` in C++.

Reply:
I  think you are over-reviewed. If you think these are problems, why you didn't propose them in the previous comments? You first think it shoud use for-range syntax, don't you see the variable "auto& name_name ", don't you see it uses  "this" ? Do you think it is funny to comment a bit once time?

Reply:
@oshynsong please note that we are multiple threaded. and we don't have an effective protocol yet to notify each other that "i have done my review".  normally, we need couple iterations to get a fix in shape.

it's fun to improve the code quality. but i don't think it's funny or anything.

> don't you see the variable "auto& name_name ", don't you see it uses "this" 

yes, i did. but i failed to publish my comment. and no, i didn't notice these indention issues last time. because i wanted to double check the format before approving a PR, and i thought your PR need more polish yesterday.